### ISSUE TITLE
`pip3 install kolibri` is broken

### ISSUE THREAD

**Issue by lyw07** [learningequality/kolibri#3164]
### Observed behavior

When users use `pip3 install kolibri` to get Kolibri 0.7.1, they are not able to run Kolibri successfully.
![screen shot 2018-02-07 at 3 27 47 pm](https://user-images.githubusercontent.com/19424916/35947185-7f96714e-0c1b-11e8-99a4-bbccc7eeceb5.png)

This is because the python2-only modules are duplicated in `kolibri/dist`. 
Below is `kolibri/dist` from the folder extracted from the whl file on **PyPi**, we can see that modules such as `html` still exist in`kolibri/dist` (and they exist in `kolibri/dist/py2only` as well):
```
Lingyi-4:dist lingyiwang$ ls -l
total 136
-rw-r--r--@  1 lingyiwang  staff    212 Feb  2 15:02 README
-rw-r--r--@  1 lingyiwang  staff      0 Feb  2 15:02 __init__.py
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:30 _dummy_thread
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:30 _markupbase
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:30 _thread
-rw-r--r--@  1 lingyiwang  staff   9316 Feb  2 15:02 args.py
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:30 builtins
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:30 certifi
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:30 cext
drwxr-xr-x  42 lingyiwang  staff   1428 Feb  7 15:30 chardet
drwxr-xr-x  29 lingyiwang  staff    986 Feb  7 15:30 cherrypy
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:30 clint
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:30 colorlog
drwxr-xr-x   4 lingyiwang  staff    136 Feb  7 15:30 concurrent
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:30 copyreg
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:30 dateutil
drwxr-xr-x  20 lingyiwang  staff    680 Feb  7 15:30 django
drwxr-xr-x  12 lingyiwang  staff    408 Feb  7 15:30 django_filters
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:30 django_js_reverse
-rw-r--r--@  1 lingyiwang  staff  19946 Feb  2 15:02 docopt.py
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:30 future
drwxr-xr-x   5 lingyiwang  staff    170 Feb  7 15:30 html
drwxr-xr-x   7 lingyiwang  staff    238 Feb  7 15:30 http
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:30 iceqube
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:30 idna
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:30 ifcfg
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:30 ipware
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:30 jsonfield
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:30 kolibri_exercise_perseus_plugin
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:30 le_utils
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:30 libfuturize
drwxr-xr-x   5 lingyiwang  staff    170 Feb  7 15:30 libpasteurize
drwxr-xr-x   7 lingyiwang  staff    238 Feb  7 15:30 metaphone
drwxr-xr-x  21 lingyiwang  staff    714 Feb  7 15:30 morango
drwxr-xr-x  17 lingyiwang  staff    578 Feb  7 15:30 mptt
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:30 past
drwxr-xr-x   4 lingyiwang  staff    136 Feb  7 15:30 porter2stemmer
drwxr-xr-x  16 lingyiwang  staff    544 Feb  7 15:30 py2only
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:30 pyasn1
drwxr-xr-x   9 lingyiwang  staff    306 Feb  7 15:30 pytz
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:30 queue
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:30 reprlib
drwxr-xr-x  20 lingyiwang  staff    680 Feb  7 15:30 requests
drwxr-xr-x  14 lingyiwang  staff    476 Feb  7 15:30 requests_toolbelt
drwxr-xr-x  40 lingyiwang  staff   1360 Feb  7 15:31 rest_framework
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:31 rest_framework_csv
drwxr-xr-x  20 lingyiwang  staff    680 Feb  7 15:31 rsa
-rw-r--r--@  1 lingyiwang  staff  30098 Feb  2 15:02 six.py
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:31 socketserver
drwxr-xr-x  25 lingyiwang  staff    850 Feb  7 15:31 sqlalchemy
drwxr-xr-x  13 lingyiwang  staff    442 Feb  7 15:31 tests
drwxr-xr-x  15 lingyiwang  staff    510 Feb  7 15:31 tkinter
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:31 tqdm
drwxr-xr-x   9 lingyiwang  staff    306 Feb  7 15:31 tzlocal
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:31 ua_parser
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:31 unicodecsv
drwxr-xr-x  15 lingyiwang  staff    510 Feb  7 15:31 urllib3
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:31 user_agents
drwxr-xr-x   3 lingyiwang  staff    102 Feb  7 15:31 winreg
drwxr-xr-x   5 lingyiwang  staff    170 Feb  7 15:31 xmlrpc
```

However, this problem doesn't exist if the whl file is from github release page. For example, if users go to our github release page to download the whl file and install kolibri with `pip3 install kolibri-0.7.1-py2.py3-none-any.whl`, the python2-only modules only exist in the `kolibri/dist/py2only` folder, which is correct.
Below is `kolibri/dist` from the folder extracted from the whl file on **github release page**, we can see that modules such as `html` don't exist in `kolibri/dist` now:
```
Lingyi-4:dist lingyiwang$ ls -l
total 136
-rw-r--r--@  1 lingyiwang  staff    212 Feb  2 06:18 README
-rw-r--r--@  1 lingyiwang  staff      0 Feb  2 06:18 __init__.py
-rw-r--r--@  1 lingyiwang  staff   9316 Feb  2 06:18 args.py
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:35 certifi
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:37 cext
drwxr-xr-x  42 lingyiwang  staff   1428 Feb  7 15:35 chardet
drwxr-xr-x  29 lingyiwang  staff    986 Feb  7 15:36 cherrypy
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:35 clint
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:36 colorlog
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:35 dateutil
drwxr-xr-x  20 lingyiwang  staff    680 Feb  7 15:36 django
drwxr-xr-x  12 lingyiwang  staff    408 Feb  7 15:35 django_filters
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:36 django_js_reverse
-rw-r--r--@  1 lingyiwang  staff  19946 Feb  2 06:18 docopt.py
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:37 future
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:37 iceqube
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:36 idna
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:35 ifcfg
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:35 ipware
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:35 jsonfield
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:37 kolibri_exercise_perseus_plugin
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:35 le_utils
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:37 libfuturize
drwxr-xr-x   5 lingyiwang  staff    170 Feb  7 15:37 libpasteurize
drwxr-xr-x   7 lingyiwang  staff    238 Feb  7 15:36 metaphone
drwxr-xr-x  21 lingyiwang  staff    714 Feb  7 15:35 morango
drwxr-xr-x  17 lingyiwang  staff    578 Feb  7 15:35 mptt
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:35 past
drwxr-xr-x   4 lingyiwang  staff    136 Feb  7 15:35 porter2stemmer
drwxr-xr-x  16 lingyiwang  staff    544 Feb  7 15:35 py2only
drwxr-xr-x   8 lingyiwang  staff    272 Feb  7 15:35 pyasn1
drwxr-xr-x   9 lingyiwang  staff    306 Feb  7 15:35 pytz
drwxr-xr-x  20 lingyiwang  staff    680 Feb  7 15:35 requests
drwxr-xr-x  14 lingyiwang  staff    476 Feb  7 15:35 requests_toolbelt
drwxr-xr-x  40 lingyiwang  staff   1360 Feb  7 15:37 rest_framework
drwxr-xr-x  10 lingyiwang  staff    340 Feb  7 15:37 rest_framework_csv
drwxr-xr-x  20 lingyiwang  staff    680 Feb  7 15:35 rsa
-rw-r--r--@  1 lingyiwang  staff  30098 Feb  2 06:18 six.py
drwxr-xr-x  25 lingyiwang  staff    850 Feb  7 15:35 sqlalchemy
drwxr-xr-x  13 lingyiwang  staff    442 Feb  7 15:35 tests
drwxr-xr-x  11 lingyiwang  staff    374 Feb  7 15:35 tqdm
drwxr-xr-x   9 lingyiwang  staff    306 Feb  7 15:35 tzlocal
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:35 ua_parser
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:36 unicodecsv
drwxr-xr-x  15 lingyiwang  staff    510 Feb  7 15:35 urllib3
drwxr-xr-x   6 lingyiwang  staff    204 Feb  7 15:35 user_agents
```

### Expected behavior

Users should be able to successfully install kolibri 0.7.1 through pip3

### User-facing consequences

Users can't install kolibri through `pip3 install`

### Errors and logs

Described in the `Observed behavior`

### Steps to reproduce

1. run `pip3 install kolibri`
2. run `kolibri start`

### Context

Kolibri version: 0.7.1
Operating system: Mac OS X

**Comment by benjaoming** [comment]
I'll check manually before releasing 0.7.2 to PyPi that it doesn't have the same problem, but need to figure out what went wrong!

**Comment by lyw07** [comment]
@benjaoming not sure if this could be possible, but I was thinking if it's caused by the `install_requires` from `setup.py`: https://github.com/learningequality/kolibri/blob/develop/setup.py#L167, which installs the dependencies (e.g. `future`) again from `install_requires` to kolibri when we release to PyPi?

**Comment by benjaoming** [comment]
@lyw07 it is peculiar! The tar.gz released on PyPi is fine (and considerably smaller in size), while only the .whl contains the duplicates. But both of them contain bundled dependencies, and `install_requires` in `setup.py` is empty both during `sdist` and `bdist_wheel`.

I have just built 0.7.2-beta2 locally, and the .whl was fine.. will try looking at the code to see if there's anything obvious from the Makefile causing this, it's important to notice that these two outputs are built right after one another, so they really should not defer.. I'm more and more suspicious that the error on PyPi is purely manual, from my side.

```
dist: setrequirements writeversion staticdeps buildconfig assets compilemessages
        pip install -r requirements/build.txt
        python setup.py sdist --format=gztar,zip --static > /dev/null # silence the sdist output! Too noisy!
        python setup.py bdist_wheel --static
        ls -l dist
```

**Comment by lyw07** [comment]
@benjaoming Thank you for your explanation! Please let me know if there's anything I can help with

**Comment by lyw07** [comment]
This issue is fixed as we have a new release 0.7.2 on PyPi
